test/meta.cpp

```diff
@@ -141,10 +141,12 @@ static_assert(std::is_same<reverse_find<L, int>, list<int, float>>::value, "");
 
 struct check_integral
 {
+    constexpr check_integral() {}
     template <class T>
-    void operator()(T &&)
+    constexpr T operator()(T && i) const
     {
         static_assert(std::is_integral<T>{}, "");
+        return i;
```

Comment:
My bad, I thought it needed a default constructor in C++11 but it actually doesn't. 
If I want to reuse it to test `for_each` I have to make it constexpr. constexpr functions can only return literal types, and void is not a literal type.